urdfenvs/tiago_reacher/resources/tiago_robot.py

```diff
@@ -14,49 +14,31 @@ def __init__(self):
         self._spawn_offset = np.array([-0.1764081, 0.0, 0.1])
 
     def set_joint_indices(self):
-        self._robot_joints = [
-            6,
-            8,
-            18,
-            19,
-            20,
-            22,
-            23,
-            24,
-            25,
-            26,
-            27,
-            28,
-            37,
-            38,
-            39,
-            40,
-            41,
-            42,
-            43,
-        ]
-        self._urdf_joints = [
-            6,
-            8,
-            19,
-            20,
-            21,
-            23,
-            24,
-            25,
-            26,
-            27,
-            28,
-            29,
-            38,
-            39,
-            40,
-            41,
-            42,
-            43,
-            44,
-        ]
-        self._castor_joints = [9, 10, 11, 12, 13, 14, 15, 16]
+        joint_info = self.get_indexed_joint_info()
+        wheel_joint_names = ["wheel_right_joint", "wheel_left_joint"] 
+        torso_joint_name = ["torso_lift_joint"] 
+        head_joint_names = ["head_" + str(i) + "_joint" for i in range(3)] 
+        arm_right_joint_names = ["arm_right_" + str(i) + "_joint" for i in range(8)] 
+        arm_left_joint_names = ["arm_left_" + str(i) + "_joint" for i in range(8)] 
+        self._joint_names = ( 
+            wheel_joint_names 
+            + torso_joint_name 
+            + head_joint_names 
+            + arm_right_joint_names 
+            + arm_left_joint_names 
+        ) 
+        robot = URDF.load(self._urdf_file) 
+        self._urdf_joints = [] 
+        for i, joint in enumerate(robot.joints): 
+            if joint.name in self._joint_names: 
+                self._urdf_joints.append(i) 
+        self._robot_joints = [] 
+        self._castor_joints = [] 
+        for index in joint_info:
+            if joint_info[index].decode("UTF-8") in self._joint_names:
+                self._robot_joints.append(index)
+            if "caster" in joint_info[index].decode("UTF-8"):
+                self._castor_joints.append(index)
```

Comment:
That doesn't look too bad. I think you understood the idea behind it.
I will comment a bit more in the comment on this PR.